frontend/screens/clubInfoView.js

```diff
@@ -5,27 +5,71 @@ By: Bruno, Hayden, Madeleine, Miriam, and Scott
 
 import React, { useState } from 'react';
 import {
-  Text, View, StyleSheet
+  Text, View, TextInput, StyleSheet, Clipboard, TouchableOpacity, AsyncStorage
 } from 'react-native';
+import { Global } from '../styles/Global';
+import clubs from '../functions/clubs';
 
+let MS_PER_DAY = 1000*60*60*24
+
+function rateMessage(end_date, progress, book) {
+	let pages_left = book.pages - progress
+	let days_left = (end_date - Date.now()) / MS_PER_DAY
+	let rate = Math.ceil(pages_left / days_left)
+	if (rate > 0) {
+		return "In order to finish by " + end_date.toDateString() + " you must read " + rate + (rate>1?" pages":" page") + " per day!"
+	}
+	else if (rate == 0) {
+		return "You have completed reading this book!"
+	}
+	else {
+		return "The target date for this club has already passed."
+	}
+}
 
 export default class ClubInfoView extends React.Component {
   constructor(props) {
+	console.log(props);
     super(props)
+	this.state = {
+		club: this.props.navigation.state.params[0],
+		book: this.props.navigation.state.params[1],
+		progress: this.props.navigation.state.params[4],
+		end: new Date(this.props.navigation.state.params[0].end*1000)
+	}
   }
-
+  
+  writeToClipboard = async () => {
+    //To copy the text to clipboard
+    await Clipboard.setString(this.state.club.clubID);
+  };
+  
+  leaveClub = () => {
+	AsyncStorage.getItem('userID').then(userID => {
+	  clubs.leave(this.state.club.clubID, userID, data => {
+		this.props.navigation.navigate('HomeScreen');
+	  })
+	})
+  }
+ 
   render() {
     return (
-      <View style={styles.container}>
-          <Text>Club Info</Text>
+      <View style={Global.container}>
+	    <View style={{flex:1, alignItems:'center'}}>
+	      <Text style={[Global.bookText, {marginTop:'15%', paddingLeft:0, textAlign:'center'}]}>{rateMessage(this.state.end, this.state.progress, this.state.book)}</Text>
+	      <Text style={[Global.bookText, {marginTop:"10%", paddingLeft:0}]}>Invite Code</Text>
+	      <TouchableOpacity onPress={this.writeToClipboard} style={Global.input}>
+		    <View>
+		      <Text style={{textAlign:"center", fontSize:24}}>{this.state.club.clubID}</Text>
+		    </View>
+	      </TouchableOpacity>
+		</View>
+		<TouchableOpacity onPress={this.leaveClub}>
+		  <View style={{height:'15%'}}>
```

Comment:
Leave club button is cut off. To fix this, the height probably needs to be changed to be a pixel value instead of a percentage value.

Reply:
Hopefully the most recent push fixed it. Let me know if things are still getting cut off.